server/src/main/java/com/scalar/db/server/DistributedStorageService.java

```diff
@@ -90,24 +98,57 @@ private static void execute(
       if (t instanceof Error) {
         throw (Error) t;
       }
+    } finally {
+      postProcess();
     }
   }
 
+  private boolean preProcess(StreamObserver<?> responseObserver) {
```

Comment:
Yes, `isPaused` was my first idea for it, actually :) I named it to `preProcess` because it will increment the outstanding request count, but `isPaused` sounds just returning the current state, and it doesn't have side effects. And I didn't come up with a good name and just named `preProcess` that is very abstract.

Do you have any good ideas?

Reply:
> I think we should rename the methods since this preProcess returns false when pause.preProcess returns true.

Ah I just realized that my intension was `pauser.preProcess` should have returned the same value as this `preProcess`. I need to fix `pauser.preProcess`.